pkg/antctl/raw/migrate/convertnetworkpolicy/command.go

```diff
@@ -0,0 +1,488 @@
+		klog.Errorf("Failed to convert Calico GlobalNetworkSet: %v", err)
+		return err
+	}
+	if err := c.convertGlobalNetworkPolicy(); err != nil {
+		klog.Errorf("Failed to convert Calico GlobalNetworkPolicy: %v", err)
+		return err
+	}
+	if err := c.convertCalicoNamespacedNetworkPolicy(); err != nil {
+		klog.Errorf("Failed to convert Calico NamespacedNetworkPolicy: %v", err)
+		return err
+	}
+	klog.Infof("Successfully converted Calico NetworkPolicy to Antrea NetworkPolicy")
+	return nil
+}
+
+func (c *converter) convertCalicoNetworkSet() error {
+	nss, err := c.calicoClient.ProjectcalicoV3().NetworkSets("").List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Failed to list Calico NetworkSets: %v", err)
+		return err
+	}
+	for _, ns := range nss.Items {
+		klog.Infof("Converting NetworkSet %s", ns.Name)
+		group := v1beta1.Group{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      ns.Name,
+				Namespace: ns.Namespace,
+			},
+			Spec: v1beta1.GroupSpec{
+				IPBlocks: []v1beta1.IPBlock{},
+			},
+		}
+
+		for _, cidr := range ns.Spec.Nets {
+			group.Spec.IPBlocks = append(group.Spec.IPBlocks, v1beta1.IPBlock{CIDR: cidr})
+		}
+		if _, err := c.antreaClient.CrdV1beta1().Groups(ns.Namespace).Create(context.Background(), &group, metav1.CreateOptions{}); err != nil {
+			klog.Errorf("Failed to create Antrea Group %s: %v", ns.Name, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *converter) convertGlobalNetworkSet() error {
+	gnss, err := c.calicoClient.ProjectcalicoV3().GlobalNetworkSets().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Failed to list Calico GlobalNetworkSets: %v", err)
+		return err
+	}
+	for _, gns := range gnss.Items {
+		klog.Infof("Converting GlobalNetworkSet %s", gns.Name)
+		clusterGroup := v1beta1.ClusterGroup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: gns.Name,
+			},
+			Spec: v1beta1.GroupSpec{
+				IPBlocks: []v1beta1.IPBlock{},
+			},
+		}
+
+		for _, cidr := range gns.Spec.Nets {
+			clusterGroup.Spec.IPBlocks = append(clusterGroup.Spec.IPBlocks, v1beta1.IPBlock{CIDR: cidr})
+		}
+		if _, err := c.antreaClient.CrdV1beta1().ClusterGroups().Create(context.Background(), &clusterGroup, metav1.CreateOptions{}); err != nil {
+			klog.Errorf("Failed to create Antrea Group %s: %v", gns.Name, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *converter) convertCalicoNamespacedNetworkPolicy() error {
+	nps, err := c.calicoClient.ProjectcalicoV3().NetworkPolicies("").List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Failed to list Calico NetworkPolicies: %v", err)
+		return err
+	}
+	for _, np := range nps.Items {
+		klog.Infof("Converting NetworkPolicy %s", np.Name)
+		anp := v1beta1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      np.Name,
+				Namespace: np.Namespace,
+				Annotations: map[string]string{
+					convertFromCalicoAnnotation: "true",
```

Comment:
what does this annotation help? Do we need it?

Reply:
This annotation shows that this NetworkPolicy was converted from Calico when users use `kubectl get` command.  Actually it will not be used in any controller.

Reply:
It's better to remove it if it's not used anywhere. OK to keep it if you need it to do some cleanups or checks during migration.